doc/lsst.pipe.base/command-line-task-data-repo-howto.rst

```diff
@@ -0,0 +1,136 @@
+.. _command-line-task-data-repo-howto:
+
+#################################################################
+Using Butler data repositories and reruns with command-line tasks
+#################################################################
+
+:ref:`Command-line tasks <using-command-line-tasks>` use Butler data repositories for reading and writing data.
+Butler data repositories are specially-configured directories on a filesystem.
```

Comment:
Excellent, this is something I wanted to bring up this was still unclear to me (I need to take another read through [LDM-463](https://ldm-463.lsst.io) and better yet, work with you on `lsst.daf.*` documentation!).

- Will the environment variable facility still work with non-POSIX repositories? Instead of a filesystem path the environment variables can be a general URI?
- Do we have a facility for deleting re-runs in non-POSIX repositories? The advice I've gotten is that folks `ls` inside the rerun directory and delete un-needed experimental reruns.

>  Ideally, no one should ever have to know how to "anoint" a directory as a repository; that is a Butler implementation detail.

Do we have a facility for creating a POSIX repository? Right now the standard pattern seems to be creating a directory and planting a `_mapper` in it. See this tutorial draft from another ticket: https://pipelines.lsst.io/v/DM-11391/getting-started/data-setup.html#creating-a-butler-repository-for-hsc-data

> Along similar lines, I wonder if this section could be renamed "Using command-line tasks to process data" and then, at the top, explain more about how/why LSST uses Butler repositories for this (and sometimes reruns).

I wanted to have the keywords/phrases "Butler data repository" and "rerun" in the title because those are keywords that (will) appear elsewhere in the documentation. To make that first connection between command-line tasks and data processing and Butler data repositories I think we might need to lean on tutorials and overview topics (that aren't in this ticket). I don't want to put too much discussion in how-to topics like this. I'm intending to write overview topics that I can link back to from the how-to.